### PR TITLE
Implement Logging

### PR DESCRIPTION
**Pull request by Joecakes4u**
progress on #244

Re-forked my repo and added updates to logs function.

### CODE REVIEW

File: vent/api/actions.py [hunk 1]
```diff
@@ -34,339 +35,350 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+    @staticmethod
+    def system_commands():
+        # restart, shutdown, upgrade, etc.
+        return
+
+    @staticmethod
+    def logs(self, grep_list):
+        """ generically filter logs stored in log containers """
+        if not grep_list:
+            grep_list = {"core","vent-management"}
```
Comment (cglewis) [inline vent/api/actions.py]: there is no more `vent-management` container in the new version :)

File: vent/api/actions.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import ast
+import docker
```
Comment (cglewis) [inline vent/api/actions.py]: I don't think this is required, since it's getting the client instance from `Plugin` which does the import, right?

File: vent/api/actions.py [hunk 3]
```diff
@@ -34,339 +34,350 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+    @staticmethod
+    def system_conf():
+        return
+
+    @staticmethod
+    def system_commands():
+        # restart, shutdown, upgrade, etc.
+        return
+
+    @staticmethod
```
Comment (cglewis) [inline vent/api/actions.py]: should remove this

File: vent/api/actions.py [hunk 4]
```diff
@@ -34,339 +34,350 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+        if not grep_list:
+            grep_list = {"core"}
+        containers = self.d_client.containers().list()
+        cores = [c for c in containers if ("core" in c.name)]
+        for expression in grep_list:
+            for core in cores:
+                # 'logs' stores each line which contains the expression
+                logs  = [log for log in core.logs().split("\n") if expression in log]
+                for log in logs:
+                    print log
```
Comment (cglewis) [inline vent/api/actions.py]: check out the logger initialized in `__init__` instead of printing:

`self.logger.info(log)`

File: vent/api/actions.py [hunk 5]
```diff
@@ -34,339 +34,350 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+    def logs(self, grep_list={}):
+        """ generically filter logs stored in log containers """
+        if not grep_list:
+            grep_list = {"core"}
+        containers = self.d_client.containers().list()
+        cores = [c for c in containers if ("core" in c.name)]
+        for expression in grep_list:
+            for core in cores:
+                # 'logs' stores each line which contains the expression
+                logs  = [log for log in core.logs().split("\n") if expression in log]
```
Comment (cglewis) [inline vent/api/actions.py]: this probably needs a try/except in case core.logs() doesn't work?

**Comment by cglewis** [comment]
otherwise looks pretty good, i think.

**Comment by Joecakes4u** [comment]
Seems there's a TravisCI error with how I called the Logger.

File: vent/api/actions.py [hunk 6]
```diff
@@ -34,339 +34,352 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+        return
+
+    def logs(self, grep_list={}):
+        """ generically filter logs stored in log containers """
+        if not grep_list:
+            grep_list = {"core"}
+        containers = self.d_client.containers().list()
+        cores = [c for c in containers if ("core" in c.name)]
+        for expression in grep_list:
+            try:
```
Comment (cglewis) [inline vent/api/actions.py]: move this inside of the `cores` loop, that way if one core breaks, it doesn't prevent the rest from logging.

File: vent/api/actions.py [hunk 7]
```diff
@@ -34,339 +34,352 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+    @staticmethod
+    def system_commands():
+        # restart, shutdown, upgrade, etc.
+        return
+
+    def logs(self, grep_list={}):
+        """ generically filter logs stored in log containers """
+        if not grep_list:
+            grep_list = {"core"}
+        containers = self.d_client.containers().list()
```
Comment (cglewis) [inline vent/api/actions.py]: looks like the travis issue with this line, which I think should just be `self.d_client.containers.list()` per the new docker API

File: vent/api/actions.py [hunk 8]
```diff
@@ -34,339 +34,352 @@ def add(self, repo, tools=None, overrides=None, version="HEAD",
...
+    @staticmethod
+    def system_conf():
+        return
+
+    @staticmethod
+    def system_commands():
+        # restart, shutdown, upgrade, etc.
+        return
+
+    def logs(self, grep_list={}):
```
Comment (cglewis) [inline vent/api/actions.py]: per the quantifiedcode issue, make `grep_list=None` instead of an empty dict.  it has an explanation of why this change is a better practice.
